app/javascript/article-form/actions.js

```diff
@@ -83,14 +83,15 @@ function generateUploadFormdata(payload) {
   return formData;
 }
 
-export function generateMainImage(payload, successCb, failureCb) {
+export function generateMainImage({ payload, successCb, failureCb, signal }) {
   fetch('/image_uploads', {
     method: 'POST',
     headers: {
       'X-CSRF-Token': window.csrfToken,
     },
     body: generateUploadFormdata(payload),
     credentials: 'same-origin',
+    signal,
   })
```

Comment:
Refactor (blocking): Since this is being updated, use the `@utiltities/http/request` utility function,
```suggestion
  request('/image_uploads', {
    method: 'POST',
    body: generateUploadFormdata(payload),
    signal,
  })
```

Reply:
The only thing is that the `request` function assumes any request body should be a JSON string, whereas this needs to be a `FormData` object - I feel like it may be the exception to using the standard `request` helper. What do you think?

Otherwise we'll need to add some refactors to the helper that would allow a body to be either FormData or JSON string

Reply:
Yeah, makes sense. We can revisit modifying `request`.